src/components/SEO/helper.ts

```diff
@@ -0,0 +1,82 @@
+import { getSrc, getImage, IGatsbyImageData } from 'gatsby-plugin-image'
+
+export type MetaProps = JSX.IntrinsicElements['meta']
+
+export const buildMetaTitle = (metaData: MetaProps[], title: string) => {
+  metaData.push(
+    {
+      property: 'og:title',
+      content: title
+    },
+    {
+      name: 'twitter:title',
+      content: title
+    }
+  )
+}
+
+export const buildMetaDescription = (
+  metaData: MetaProps[],
+  description: string
+) => {
+  metaData.push(
+    {
+      name: 'description',
+      content: description
+    },
+    {
+      property: 'og:description',
+      content: description
+    },
+    {
+      name: 'twitter:description',
+      content: description
+    }
+  )
+}
+
+export const buildMetaImage = (
```

Comment:
I don't think it's the best practice to have these functions directly mutate `prebuiltMeta`. Maybe just have functions like `buildMetaImage` just return the meta data itself and keep the `prebuildMeta` inside of `buildMetadata`?

Example:
```
// etc...

export const getMetaImage = (
  imageUrl: string,
  imageAlt?: string,
  imageData?: IGatsbyImageData
): Array<MetaProps> => {
  return [
    { property: 'og:image', content: imageUrl },
    { name: 'og:image:alt', content: imageAlt },
    { property: 'og:image:width', content: String(imageData?.width) },
    { property: 'og:image:height', content: String(imageData?.height) },
    { name: 'twitter:image', content: imageUrl },
    { name: 'twitter:image:alt', content: imageAlt }
  ]
}

export const buildMetadata = (
  siteUrl: string,
  title?: string,
  defaultMetaTitle?: boolean,
  description?: string,
  keywords?: string,
  image?: IGatsbyImageData,
  imageAlt?: string
) => {
  const prebuildMeta: MetaProps[] = []

  // etc...

  if (image) {
    const imageUrl = siteUrl + getSrc(image)
    const imageData = getImage(image)
    prebuildMeta.push(...getMetaImage(imageUrl, imageAlt, imageData))
  }
  return prebuildMeta
}

```

Reply:
I totally agree with you! using functional programming approach looks much better for this scenario to avoid mutating the array or returning a new one <3